### DISCUSSION TITLE
Does not build with the default setup on M2

### DISCUSSION THREAD

**Discussion by Subhadiptech** [oppia/oppia-android#4721] [Q&A (Installation)]
The project does not build with the default setup.
Running macOS Ventura M2 building for pixel3a API 29

Following the setup for mac [here](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android)
Install oppia-android and Run the app from Android Studio.

`Execution failed for task ':model:generateProto'.

Could not resolve all files for configuration ':model:protobufToolsLocator_protoc'.
Could not find protoc-3.8.0-osx-aarch_64.exe (com.google.protobuf:protoc:3.8.0).
Searched in the following locations:
[https://repo.maven.apache.org/maven2/com/google/protobuf/protoc/3.8.0/protoc-3.8.0-osx-aarch_64.exe`](https://repo.maven.apache.org/maven2/com/google/protobuf/protoc/3.8.0/protoc-3.8.0-osx-aarch_64.exe%60)

Expected to build without issue.

The issue seems like the same one mentioned here https://github.com/oppia/oppia-android/issues/4644
There was a solution of adding proto to local.properties but it should be contained in documentations so that user can easily setup the project locally.

**Comment by BenHenning** [comment]
Thanks for reporting this @Subhadiptech.

Moving this over to be a discussion so that we can investigate where the gap is here.

**Reply by BenHenning** [reply]
@adhiamboperes given that you investigated this, do you have any thoughts here?

@Subhadiptech I agree with us potentially needing to update the setup instructions, but what isn't clear to me: does updating local.properties fix your issue, or is there still a problem?

**Reply by adhiamboperes** [reply]
I have had a quick look at the setup scripts, and the current configuration should work even for M2 Macs.

 One thought I have is, @Subhadiptech, did you run `bash scripts/setup.sh` in Rosetta? Failure to do so causes `scripts/buf_m1_mac_setup.sh` to not be executed.

`buf_m1_mac_setup.sh` is what should create `protobuf_platform=osx-x86_64 ` in `local.properties`, and you should not need to do it manually.

You can revisit the setup instructions [here](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android)

**Reply by adhiamboperes** [reply]
> @Subhadiptech I agree with us potentially needing to update the setup instructions, but what isn't clear to me: does updating local.properties fix your issue, or is there still a problem?

Echoing @BenHenning's question, as this could point to whether the setup scripts executed correctly.

**Comment by Subhadiptech** [comment]
@adhiamboperes I have not ran it on rosetta. Let me give it a check.

**Reply by Subhadiptech** [reply]
<img width="566" alt="Screenshot 2022-11-19 at 5 12 02 PM" src="https://user-images.githubusercontent.com/89024718/202849014-902afadb-1bd8-4dd4-848d-f0051c6f4e83.png">

I ran this on rosetta

**Reply by adhiamboperes** [reply]
I believe you should now be able to deploy the app without any issues. Is that correct?

**Reply by Subhadiptech** [reply]
No still I am not able to do.

**Reply by adhiamboperes** [reply]
@Subhadiptech, it would be great if you could  specify whether you are still getting the original error message, or a new one. Could you also address @BenHenning's question?

**Comment by MohitGupta121** [comment]
@Subhadiptech you can try solution in this [disscussion](https://github.com/oppia/oppia-android/discussions/4526) this might helps you.

**Reply by Subhadiptech** [reply]
Sure

**Reply by MohitGupta121** [reply]
@Subhadiptech if it's work for you please mark this as answered.

**Reply by deonwaju** [reply]
This solution worked for me as well

**Reply by MohitGupta121** [reply]
@Subhadiptech Is still you getting that error or is fixed? 
If it fixed please mark this as answer so that it's helps to other.
